topics/metagenomics/tutorials/mothur-miseq-sop/content.md

```diff
@@ -256,7 +256,7 @@ the reads for all the overlapping positions.
 
 > ### {% icon hands_on %} Hands-on: Combine forward and reverse reads into contigs
 >
-> - **Make.contigs** {% icon tool %} with the following parameters
+> -  {% tool [Make.contigs](toolshed.g2.bx.psu.edu/repos/iuc/mothur_make_contigs/mothur_make_contigs/1.39.5.1) %} with the following parameters
```

Comment:
I think these might be using the latest versions of the tools? @shiltemann do you prefer that, or that we stick to the versions that are written in the workflow (which we reasonably know works)

Reply:
it should work with these versions, the workflow just needs to be updated (something else I've been meaning to do forever)

Reply:
Will that cause any errors in the history ?

Other than that i have faced three problems while doing this training, in workflow 5, 6 and in the make.biom step which shows empty output so i couldnot proceed with the phinch tool.

<img width="787" alt="Workflow 5" src="https://user-images.githubusercontent.com/7595309/120168556-48f88580-c1ff-11eb-8910-33cb4fad5a5e.png">
<img width="1018" alt="Workflow 6" src="https://user-images.githubusercontent.com/7595309/120168563-4bf37600-c1ff-11eb-9969-f89fe8bc2757.png">

Reply:
Thanks for letting me know @EngyNasr, the group to remove in your first screenshot should be "Mock", but it should not have to be chosen by the user, it should be set in the workflow itself, hmm.

One thing we could do is run through the long tutorial step by step (so not using the workflows) with the latest versions of the tool and see where we run into problems. Once we successfully complete the tutorial manually, we can extract the workflows from our history. But this is a bit more work so I can also help with that if needed.

Reply:
Great so I will perform the extended one, and note things down

Reply:
So now everything worked so fine in the extended version, except for the very last step of the make.biom, I got an error.

Here is the history of the extended version: https://usegalaxy.eu/history/export_archive?id=9f5c448f02a34e33&jeha_id=196d4a5f031e1451

Reply:
I'm getting a "History not accessible by user" message for that history, could you publish the workflow and send me the link?

Also: thanks for testing, that was fast, I know it's a really long tutorial! glad to hear most of it still works with latest version of the tools, I will have a look at the biom step

Reply:
Hey Saskia :), you are more than welcome, i also guess that my problem with the short version in workflow 5, was that I havenot selected the group "mock" as a parameter in the Remove.groups step. So just guide me through updating the workflows and I can help in that too.

Regarding the history, I think I did the extraction wrong, so here you go:  https://usegalaxy.eu/u/engy.nasr/h/16s-mianalysis-mothur-extended

Reply:
@shiltemann shall i do something for now for updating the workflow ?

Reply:
Thanks! I will check tomorrow to see if I can find out what is going wrong with the biom step. Once we have it working you can extract the full workflow from your history for the long version of the tutorial, and the smaller workflows for the short tutorial (you could probably already do the first 4 of those now, since the problem is at the end)

Let me know if you need some help extracting workflows from a history?

Reply:
ok, I think the problem occurred at the very beginning, when creating the collection of input datasets:

![image](https://user-images.githubusercontent.com/2563865/121163079-c8fd9b80-c84e-11eb-9a85-61a1881c1e1e.png)

The sample names here should be without the `_R` behind them. The problem later comes because the make.biom tool matches these names to the sample names in the metadata file.

It would be too much work to do the whole thing again, so I would say just extract the workflow from this history, it should still be fine, since the step that makes the collection isn't part of the workflow. Here are instructions for extracting the workflow from your history: https://training.galaxyproject.org/training-material/faqs/galaxy/workflows_extract_from_history.html. (you can leave the failed make.biom step in your history in this case, since we know that the settings are correct, only the input data had a problem).

When you have the workflow, you can test is if you like with a new and correct collection, and my guess is it should run just fine. 

After that, you make the smaller workflows for the shorter tutorial in the same way, but only including some of the steps when you extract the workflow from your history (or if you prefer, you can do this using the workflow editor, make a copy of the full workflow, and delete all the steps except the ones for that section)

Let me know if you need any help doing this, just having the full workflow would already be a big help, thanks!

Reply:
Thanks a lot for the explanation, I will follow all your comments and will let you know directly whenever I am stuck :)

Reply:
@shiltemann Heyy :) I have done all your suggestions and just finished testing the updated extracted workflows in the training short version. All outputs are now green with the desired output. Even the make.biom worked :) here is the full workflow of the updated short version of the training ( https://usegalaxy.eu/u/engy.nasr/w/workflow-constructed-from-history-16s-microbial-analysis-with-mothur-short-updated )

The following is what is done: 
Updated to the latest versions of tools, e.g. Updating Screen.seqs tool in the Workflows from version 1.39.5.0 to 1.39.5.1

Added missing arguments in:
 - Workflow 5, remove group tool, groups: mock

 - Workflow 7, venn, groups: F3D0, F3D1, F3D2, F3D3

I will now add the updated workflows to the training (short version) and commit it. 

Please let me know if I can update or fix anything

Reply:
awesome @EngyNasr! I will review this today or tomorrow and get it merged, it's great to have this for the upcoming GCC conference, thanks a lot!

Reply:
you are more than welcome, actually thanks to you and @bebatut for teaching me :))

Reply:
Hey @shiltemann :)), did it work with you as well? Please let me know if I need to update anything else. It would be great if we can merge before next week. Thank u

Reply:
Hey @EngyNasr sorry, have not tried it yet, but will try to do that soon. 

Did you also add the new version of the full workflow to the extended tutorial? Or was that one already up-to-date?

Reply:
Hey Hey :)) Thank you so much :) 
I have done the extended version, it doesn't include a workflow, however it includes the tools, I will revise the tools' links that I added there, and make sure that they are all in their latest versions.
Also please let me know if I am missing something, or if there is a workflow that I didn't see.

Thanks a lot

Reply:
So just finished checking the extended version, in the PR all tools' links are in their latest version